### PR TITLE
Remove bootstrap

### PR DESCRIPTION
**Pull request by bunnymatic**
Problem
-------

Bootstrap, though it can help to get things rolling, it is not ideal in
the modern day where we often move rapidly to a Javascript
library/framework that has CSS-in-JS or some other scheme to manage
styling.

Including bootstrap includes (implicitly) it's own reset CSS that is
fine at first.  But if you later decide to remove Bootstrap, you may
find that your components no longer look right because you've lost that
implicit reset.

Solution
--------

Remove bootstrap

Changes
-------

* Remove bootstrap from package.json
* update the stylesheets we have to be a bit simpler but still rendering
a header and footer

Screenshots (optional):
-----------------------
Main screenshot

Screen Shot 2019-12-11 at 11 50 08 AM

Imagine we have a model called `Thing` with attribute `name`, here are the scaffolded templates.

Screen Shot 2019-12-11 at 11 50 35 AM
Screen Shot 2019-12-11 at 11 50 45 AM
Screen Shot 2019-12-11 at 11 50 51 AM

### CODE REVIEW

**Comment by bunnymatic** [comment]
In preparation for the question "What do we do for back-office/quick-start pages if we have no real css framework/story?", I have the following.

As mentioned above, one major drawback of adding Bootstrap at this stage is that it includes an opinionated reset/normalize.  If development is done with this in place, and later Bootstrap is removed, extra work will be required to replace/re-engineer that reset.

Additionally I think that CSS frameworks to solve this back-office or quick-start problem (like any of the following: [Ant Design](https://ant.design/), [MaterialUI](https://material-ui.com/), [Bootstrap](https://getbootstrap.com/), [Foundation], [Spectre](https://picturepan2.github.io/spectre/), [PureCSS](https://purecss.io/) are relatively easy to add after the fact.  

As an organization, we used to leverage Bootstrap far more than I have seen over the past few years.  More recently I've seen migration to frameworks that are more in tune with Angular or React projects which require a different kind of inclusion (often it's more like a component library that provides CSS with some kind of CSS in JS solution).

Given that move, I believe that leaving Bootstrap out of this template and letting the team choose their framework of choice based on the projects design constraints is better.

**Comment by eskfung** [comment]
I love the thrust of this PR. Would you want to highlight the aforementioned CSS frameworks in our Readme to give more guidance for the follow-up steps?

File: app/views/layouts/application.html.slim [hunk 1]
```diff
@@ -8,31 +8,13 @@ html
...
-              / - else
-              /   a.nav-link href="/" Sign In
-
+    header.header
+      .container
+        a.header__home href="/" App Prototype
 
     main.container role="main"
       - flash.each do |name, msg|
         .alert.alert-dismissable class=alert_class(name) role="alert"
```
Comment (mattbrictson) [inline app/views/layouts/application.html.slim]: Should we drop `.alert-dismissable` and the `alert_class` helper? It doesn't seem like there are any style definitions for those classes.

File: app/javascript/stylesheets/_colors.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+$white: #ffffff;
+$gray-40: #404040;
+$gray-99: #999999;
+$gray-cc: #cccccc;
+$gray-fa: #fafafa;
+$black: #000000;
```
Comment (mattbrictson) [inline app/javascript/stylesheets/_colors.scss]: 👍

File: app/views/layouts/application.html.slim [hunk 3]
```diff
@@ -8,31 +8,13 @@ html
...
-              /       = current_user.email
-              /
-              /     .dropdown-menu aria-labelledby="dropdown"
-              /       a.dropdown-item href="/" Sign Out
-              / - else
-              /   a.nav-link href="/" Sign In
-
+    header.header
+      .container
+        a.header__home href="/" App Prototype
```
Comment (mattbrictson) [inline app/views/layouts/application.html.slim]: `.header__home` doesn't seem to be defined. Do we need it here?

File: app/views/layouts/application.html.slim (hunk 1 above)
Comment (bunnymatic) [reply app/views/layouts/application.html.slim]: sure.  i debated about removing this entirely, but the notion of having a place to put an alert seemed like a nice thing.  so ... i'll dump the extra classes.

**Comment by bunnymatic** [comment]
@eskfung @mattbrictson I agree about trying to answer the "what if i want bootstrap?" 
The reason I haven't yet added it to the readme was that I wondered about how many of those cases do we want to support?  
> What if I want PureCSS/Foundation/MaterialUI/SuperCoolNewCSSThing?

I'm fine with adding notes just for Bootstrap since it'll be pretty easy (cuz I just ripped it out) and because it's the thing we're removing here.  Do you think that's enough?  Or is there any reason to add a couple answers to "What if i want <framework x>?" that should also be added here.

**Comment by mattbrictson** [comment]
> I'm fine with adding notes just for Bootstrap since it'll be pretty easy (cuz I just ripped it out) and because it's the thing we're removing here. Do you think that's enough?

I personally think that is enough. I think documenting the other frameworks is a bit of scope creep for this PR.

**Comment by bunnymatic** [comment]
We discussed where to put the notes about "what if i want bootstrap?" and agreed that we will add a "next steps" section in the readme on the carbonfive/raygun repo.  There we can suggest ways for folks to get bootstrap easily into the project after it's been zapped by raygun.

Additionally, we'll add notes to that effect in the `print_next_steps_carbon_five` method also in that repo.

**Comment by bunnymatic** [comment]
This will get merged in concert with https://github.com/carbonfive/raygun/pull/141

**Pull request merged**